AEXS/aex-nft.md

```diff
@@ -0,0 +1,413 @@
+| token_id | int |
+| enabled | bool |
+
+## *ApprovalForAll*
+
+This event MUST be triggered and emitted upon a change of operator status, including revocation of approval.
+
+The event arguments should be as follows: `(owner, operator, approved)`
+
+```sophia
+ApprovalForAll(indexed address, indexed address, bool)
+```
+
+| parameter | type |
+| :--- | :--- |
+| owner | address |
+| operator | address |
+| approved | bool |
+
+## Extension Mintable ("mintable")
+
+### mint\(\)
+
+Issues a new token to the provided address. If the `owner` is a contract, NFTReceiver will be called with `data` if provided.
+Emits a Transfer event.
+Throws if the call to NFTReceiver implementation failed (safe transfer). 
+
+```sophia
+stateful entrypoint mint : (owner address, token_data option(metadata), data option(string)) => int
+```
+| parameter | type | 
+| :--- | :--- | 
+| owner | address |
+| token_data | option(metadata) |
+| data  | option(string) | 
+
+| return | type |
+| :--- | :--- |
+| token id | int |
+
+## Extension Burnable ("burnable")
+
+### burn\(\)
+
+This function burns the token of the provided token id from `Call.caller`. Triggers the `Transfer` event.
+
+```sophia
+stateful entrypoint burn(token: int) : unit
+```
+
+| parameter | type |
+| :--- | :--- |
+| token | int |
+
+## Extension Swappable ("swappable")
+
+### swap\(\)
+
+This function burns the whole balance of the `Call.caller` and stores the same amount in the `swapped` map. 
+
+```sophia
+stateful entrypoint swap() : unit
+```
+
+| parameter | type |
+| :--- | :--- |
+| value | int |
+
+| return | type |
+| :--- | :--- |
+| () | unit |
+
+### check_swap\(\)
+
+This function returns the amount of tokens that were burned trough `swap` for the provided account. 
+
+```sophia
+stateful entrypoint check_swap(account: address) : int
+```
+
+| parameter | type |
+| :--- | :--- |
+| account | address |
+
+| return | type |
+| :--- | :--- |
+| int | int |
+
+### swapped\(\)
+
+This function returns all of the swapped tokens that are stored in contract state. 
+
+```sophia
+stateful entrypoint swapped() : map(address, int)
+```
+
+| return | type |
+| :--- | :--- |
+| swapped | map(address, int) |
+
+## Events
+
+**Swap** - MUST trigger when tokens are swapped using the `swap` function.
+
+The swap event arguments should be as follows: `(account,  value)`
+
+```sophia
+Swap(address, int)
+```
+
+| parameter | type |
+| :--- | :--- |
+| account| address |
+| value | int |
+
+# Receiver contract interface
+
+## NFTReceiver
+
+```sophia
+contract interface NFTReceiver = 
+    entrypoint on_nft_received : (address, address, int, option(string)) => unit
+
+### on_nft_received\(\)
+
+Deals with receiving NFT tokens on behalf of a smart contract. Contracts receiving NFT tokens should implement this interface. Mint and transfer transactions will invoke the `on_nft_received` function.
+
+```sophia
+entrypoint on_nft_received : (address, address, int, option(string)) => unit
+```
```

Comment:
I think something went wrong with the old suggestion. I also prefer `NFTs` over `NFT tokens` as the "T" already stands for token.

```suggestion
Deals with receiving NFTs on behalf of a smart contract. A smart contract MUST implement this interface to be able to receive NFTs. Mint and transfer transactions will invoke the `on_nft_received` function.

```sophia
contract interface NFTReceiver = 
    entrypoint on_nft_received : (address, address, int, option(string)) => unit
```
```